src/context.rs

```diff
@@ -668,6 +756,92 @@ mod tests {
         }
     }
 
+    mod before_all {
```

Comment:
By the way, from this code you can see how to test `after_all` I think. I can explain to you over a chat if you need further explanations. 😄